### PR TITLE
hide operator API

### PR DESCRIPTION
**Pull request by typhoonzero**

### CODE REVIEW

**Review by reyoung — DISMISSED** [review]

File: paddle/fluid/pybind/protobuf.cc [hunk 1]
```diff
@@ -301,7 +301,7 @@ void BindOpDesc(pybind11::module *m) {
              std::string ser(seriralized);
              self.SetAttr(name, ser);
            })
-      .def("block_attr", &pd::OpDesc::GetBlockAttr)
+      .def("_block_attr", &pd::OpDesc::GetBlockAttr)
```
Comment (gongweibao) [inline paddle/fluid/pybind/protobuf.cc]: _block_attr=>_block_attr_id ?
GetBlockAttr => GetBlockAttrId?

File: paddle/fluid/pybind/protobuf.cc [hunk 2]
```diff
@@ -286,12 +286,12 @@ void BindOpDesc(pybind11::module *m) {
...
-      .def("rename_output", &pd::OpDesc::RenameOutput)
+      .def("_rename_input", &pd::OpDesc::RenameInput)
+      .def("_rename_output", &pd::OpDesc::RenameOutput)
       .def("has_attr", &pd::OpDesc::HasAttr)
       .def("attr_type", &pd::OpDesc::GetAttrType)
       .def("attr_names", &pd::OpDesc::AttrNames)
-      .def("set_attr", &pd::OpDesc::SetAttr)
+      .def("_set_attr", &pd::OpDesc::SetAttr)
       .def("attr", &pd::OpDesc::GetAttr)
       .def("set_block_attr", &pd::OpDesc::SetBlockAttr)
```
Comment (gongweibao) [inline paddle/fluid/pybind/protobuf.cc:295]: set_block_attr暴露出来以后，为何block_attr要隐藏？

**Review by gongweibao — CHANGES_REQUESTED** [review]

File: python/paddle/fluid/backward.py [hunk 3]
```diff
@@ -326,7 +326,7 @@ def _append_backward_ops_(block,
         grad_sub_block_list = []
         # If the op has its own sub-block, deal with the sub-block first
         if op.has_attr("sub_block"):
-            sub_block = program.block(op.block_attr("sub_block"))
+            sub_block = program.block(op._block_attr("sub_block"))
```
Comment (gongweibao) [inline python/paddle/fluid/backward.py]: Use a protected class method?

**Review by gongweibao — DISMISSED** [review]
I'll fix them

**Review by panyx0718 — DISMISSED** [review]

**Review by panyx0718 — DISMISSED** [review]

**Review by reyoung — DISMISSED** [review]

**Review by qingqing01 — DISMISSED** [review]

**Review by qingqing01 — APPROVED** [review]

**Review by panyx0718 — APPROVED** [review]

**Pull request merged**